go.mod

```diff
@@ -9,7 +9,7 @@ require (
 	github.com/celestiaorg/celestia-app v0.2.0-rc-1
 	github.com/celestiaorg/go-libp2p-messenger v0.1.0
 	github.com/celestiaorg/nmt v0.8.0
-	github.com/celestiaorg/rsmt2d v0.3.1
+	github.com/celestiaorg/rsmt2d v0.4.0
```

Comment:
Should we bump this to 0.5.0 directly?

Reply:
I tried, and there are some issues with it. Specifically, it makes the code even bulkier. We discussed this with John and planned to create a few more issues in the rsmt2d repo to improve this.